### PR TITLE
webhooks: create new namespace for incoming + outgoing

### PR DESCRIPTION
**Pull request by courier-new**
This PR performs a light structural refactor to make way for docs for [outbound webhooks](https://github.com/sourcegraph/sourcegraph/issues/38278). We've been using just "webhooks" to describe inbound webhooks in a lot of places, because that's the only direction of webhooks we've supported up until 5.0. This PR helps make the distinction a bit clearer by:

- Consolidating a new site admin sidebar section for both types of webhooks ([#ask-design thread](https://sourcegraph.slack.com/archives/C0HMGV90V/p1679004219020309))
  image
- Updating the web client page routes from `site-admin/webhooks` to `site-admin/webhooks/incoming`
- Updating the docsite route from `config/webhooks.md` to `config/webhooks/incoming.md`

I also cleaned up a few things with page titles, headers, buttons, etc. to make the incoming and outgoing webhook pages a bit more visually consistent with each other, and then updated the screenshots in the docs. I did _not_ clean up component names or implementations, file directory structure, references to webhooks in backend code, or anything else internal that's inconsistent in order to keep this PR small and low-risk enough to confidently backport to 5.0.

I noticed while working on this that we had marked that we intended to remove the legacy webhook logs stuff in this version. Obviously it's a bit late to do that now, but the last thing I did in this PR was update the old web client page route to redirect straight to the new one now (see [this commit](https://github.com/sourcegraph/sourcegraph/pull/49570/commits/9adf0e5cdff8c52eb8bc9a596003cf68078ee63f)), to bring us one step closer to removing that legacy stuff! 😄

## Test plan

Manual testing of site admin routes. UI changes have Storybook story coverage.

### CODE REVIEW

**Comment by courier-new** [comment]
Hey @sashaostrikov, I tagged you here because I know you worked on the overhaul for internal webhooks, so I wanted to know if you had any other thoughts about this! But if you're not the right person for this right now, I'd be happy to loop in someone else from repo management, just let me know. 🙂

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.12% (+3.35 kb) | -0.19% (-28.00 kb) 🔽 | -0.27% (-31.35 kb) 🔽 | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/kr/reorg-site-admin-webhooks/compare-report.html;tab=live_object) for a full comparison between the commits 9adf0e5 and e9b9ef1 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by sashaostrikov** [comment]
@courier-new that's me! :) I'll take a look at your PR soon, thanks for creating it 🙏🏻

**Review by sashaostrikov — APPROVED** [review]
Thank you so much for doing this!

I've ran it locally, found a couple of issues which I decided to [fix](https://github.com/sourcegraph/sourcegraph/pull/49570/commits/7215f99a2f8ac0d2044398b53f1cc5e035b38954) right away to tighten the feedback loop and so you don't waste your time on that, I hope it is fine :)

Everything is working good, all the docs are fine, no 404s whatsoever, nice migration!

The only thing left, **which is not blocking**, is a design+similarity thing, PTAL at screenshot with details:

CleanShot 2023-03-17 at 2 01 14@2x

**Review by BolajiOlajide — APPROVED** [review]

File: client/web/src/site-admin/SiteAdminWebhookCreatePage.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { FC, useEffect } from 'react'
 
-import { mdiCog } from '@mdi/js'
+import { mdiWebhook } from '@mdi/js'
```
[the remark this answers is not in the document]
Comment (courier-new) [reply client/web/src/site-admin/SiteAdminWebhookCreatePage.tsx:3]: I was also pleased to discover it! 😄

**Comment by courier-new** [comment]
> The only thing left, which is not blocking, is a design+similarity thing, PTAL at screenshot with details:

All good points. I feel like the "correct" thing to do here is refactor to a shared set of components that could be leveraged on both pages, rather than continually tweaking one or the other to bring them closer in alignment to each other, so I've filed https://github.com/sourcegraph/sourcegraph/issues/49650 to properly follow up on this. 🙂

**Pull request merged**